### PR TITLE
Branch add graded test

### PR DESCRIPTION
**Pull request by raydenlim**
Add GradedTest Ui
Add GradedTest style
Update UG Link

### CODE REVIEW

**Comment by raydenlim** [comment]
Fix checkStyle error
Remove unnecessary comments

**Comment by raydenlim** [comment]
Add Test Cases for `GradedTestList`
Add Test Cases for `GradedTestListBook`
Add Test Cases for `GradedTestNameContainsKeywordsPredicateTest`

**Comment by raydenlim** [comment]
Add `EditGradedTestCommand`
Add `EditGradedTestCommandTest`
Add `EditGradedTestCommandParser`
Add `EditGradedTestCommandParserTest`
Update `Ui`
Remove random print statements in `StorageManagerTest`

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java [hunk 1]
```diff
@@ -48,6 +53,45 @@ public EditCommand parse(String args) throws ParseException {
...
+        }
+        if (argMultimap.getValue(PREFIX_FINALS).isPresent()) {
+            editPersonDescriptor.setFinals(
+                    ParserUtil.parseFinals(argMultimap.getValue(PREFIX_FINALS).get()));
+        }
+        if (argMultimap.getValue(PREFIX_PRACTICAL_EXAM).isPresent()) {
+            editPersonDescriptor.setPracticalExam(
+                    ParserUtil.parsePracticalExam(argMultimap.getValue(PREFIX_PRACTICAL_EXAM).get()));
+        }
+    }
```
Comment (wesho1107) [inline src/main/java/seedu/address/logic/parser/EditCommandParser.java:91]: Good abstraction here!

File: src/main/java/seedu/address/logic/parser/EditGradedTestCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                practicalExam = ParserUtil.parsePracticalExam(argMultimap.getValue(PREFIX_PRACTICAL_EXAM).get());
+            } else {
+                // If the segment doesn't match any prefix, try to parse it as an index
+                try {
+                    index = ParserUtil.parseIndex(segment);
+                } catch (ParseException ignored) {
+                    // Ignore and continue parsing other segments
+                }
+            }
+        }
```
Comment (wesho1107) [inline src/main/java/seedu/address/logic/parser/EditGradedTestCommandParser.java:79]: Would a switch case be better here?

File: src/test/java/seedu/address/logic/commands/EditGradedTestCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+                targetIndex,
+                Optional.of(new ReadingAssessment1("10.0")),
+                Optional.of(new ReadingAssessment2("20.0")),
+                Optional.of(new MidTerms("70.0")),
+                Optional.of(new Finals("100.0")),
+                Optional.of(new PracticalExam("100.0")));
+
+        // Test for equality
+        assertTrue(editGradedTestCommandWithRA1.equals(editGradedTestCommandWithRA1));
+        assertTrue(editGradedTestCommandWithRA1.equals(new EditGradedTestCommand(
```
Comment (wesho1107) [inline src/test/java/seedu/address/logic/commands/EditGradedTestCommandTest.java:182]: Consider using assertEquals

**Review by wesho1107 — APPROVED** [review]
Other than a few nits and suggestions, LGTM!

File: src/test/java/seedu/address/logic/commands/EditGradedTestCommandTest.java (hunk 3 above)
Comment (raydenlim) [reply src/test/java/seedu/address/logic/commands/EditGradedTestCommandTest.java:182]: rayson said if we use assertEqual, the codecov cant catch the coverage or smt haha

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java (hunk 1 above)
Comment (raydenlim) [reply src/main/java/seedu/address/logic/parser/EditCommandParser.java:91]: <3

**Pull request merged**